### PR TITLE
test: add more glados agents tests

### PR DESCRIPTION
**Pull request by matthiasn**
## Summary by CodeRabbit

* **Tests**
  * Added comprehensive property-based tests for improver, project, and task agent creation paths.
  * Tests exercise many scenario combinations and assert correct eligibility checks, error handling, created agent identity, state upserts, link writes, scheduling/wake behavior, subscriptions, and side-effect/no-op behavior when invariants fail, improving confidence across edge cases.

### CODE REVIEW

File: test/features/agents/service/task_agent_service_test.dart [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import 'package:flutter_test/flutter_test.dart';
+import 'package:glados/glados.dart' as glados;
 import 'package:lotti/features/agents/model/agent_config.dart';
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/agents/service/task_agent_service_test.dart:3]: Done. Imported `agent_constants.dart`.

File: test/features/agents/service/task_agent_service_test.dart [hunk 2]
```diff
@@ -79,6 +256,272 @@ void main() {
...
+            ],
+            _GeneratedTaskTemplateSlot.defaultNone => <AgentTemplateEntity>[],
+            _ => <AgentTemplateEntity>[],
+          };
+        });
+        when(
+          () => generatedRepository.getLinksTo(
+            taskId,
+            type: 'agent_task',
+          ),
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/agents/service/task_agent_service_test.dart:366]: Done. Replaced with `AgentLinkTypes.agentTask`.

File: test/features/agents/service/task_agent_service_test.dart [hunk 3]
```diff
@@ -79,6 +256,272 @@ void main() {
...
+        final createCall = verify(
+          () => generatedAgentService.createAgent(
+            kind: captureAny(named: 'kind'),
+            displayName: captureAny(named: 'displayName'),
+            config: captureAny(named: 'config'),
+            allowedCategoryIds: captureAny(named: 'allowedCategoryIds'),
+          ),
+        ).captured;
+        expect(createCall[0], 'task_agent', reason: '$scenario');
+        expect(
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/agents/service/task_agent_service_test.dart:452]: Done. Replaced with `AgentKinds.taskAgent`.
